nipype/algorithms/confounds.py

```diff
@@ -323,21 +323,35 @@ class CompCorInputSpec(BaseInterfaceInputSpec):
                               desc=('Position of mask in `mask_files` to use - '
                                     'first is the default.'))
     components_file = traits.Str('components_file.txt', usedefault=True,
-                           desc='Filename to store physiological components')
+                                 desc='Filename to store physiological components')
     num_components = traits.Int(6, usedefault=True) # 6 for BOLD, 4 for ASL
-    use_regress_poly = traits.Bool(True, usedefault=True,
+    use_regress_poly = traits.Bool(True, usedefault=True, xor=['high_pass_filter'],
```

Comment:
After adding the `xor` entries, in order to use `high_pass_filter`, we need to turn off `usedefault` here, which means we could be breaking existing scripts in one way or another.

Given that, maybe the better way to do this is to switch to an `Enum`, such as

```Python
pre_filter=traits.Enum('polynomial', 'high_pass', False, usedefault=True,
                       desc='Detrend time series prior to component extraction')
```

WDYT?